### PR TITLE
[SPARK-12534][DOC] update documentation to list command line equivalent to properties

### PR DESCRIPTION
**Pull request by felixcheung**
Several Spark properties equivalent to Spark submit command line options are missing.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -120,7 +120,8 @@ of the most common options to set are:
   <td><code>spark.driver.cores</code></td>
   <td>1</td>
   <td>
-    Number of cores to use for the driver process, only in cluster mode.
+    Number of cores to use for the driver process, only in cluster mode. This can be set through
+    <code>--driver-cores</code> command line option.
```
Comment (srowen) [inline docs/configuration.md]: I don't think the purpose of this file is to document how the CLI works. It should stick to documenting the underlying properties.

Comment (felixcheung) [inline docs/configuration.md]: I moved these to the running-on-yarn doc, would that work?

**Comment by srowen** [comment]
I guess my point is that these CLI flags are already documented in the CLI, as help text. 

I can see an argument for formally, separately documenting them in the website, and/or their relation to other Spark properties if applicable. It's duplicative and means keeping them in sync, but it's useful. However you'd want to do that for all or none, I suppose, rather than just a few. Right now I don't see any existing place for them. 

In bits and pieces, the flags are referenced in various parts of the docs. My suggestion would be to not do this at all, or else, find where similar arguments are already mentioned in the docs and mention these few additional ones there. For example `executor-cores` is mentioned a little in `job-scheduling.md`

**Comment by felixcheung** [comment]
Thank you for the detailed explanation, @srowen.
Agreed it would be valuable to have a dedicated doc page for CLI flags, perhaps as a bigger project later. As of now, I have moved references to config prop to job-scheduling.md. I hope that looks alright.

**Comment by srowen** [comment]
Looks reasonable; while we're here, does `running-on-yarn.md` need to document `spark.driver.memory` for consistency? I see `spark.driver.cores` documented.

**Comment by felixcheung** [comment]
good idea, added spark.driver.memory

**Comment by srowen** [comment]
LGTM

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**